source/common/network/raw_buffer_socket.h

```diff
@@ -25,8 +25,10 @@ class RawBufferSocket : public TransportSocket, protected Logger::Loggable<Logge
   bool startSecureTransport() override { return false; }
   void configureInitialCongestionWindow(uint64_t, std::chrono::microseconds) override {}
 
-private:
+protected:
   TransportSocketCallbacks* callbacks_{};
```

Comment:
I'm not seeing where this is used. can this be reverted or is it used downstream?

If not, while Envoy isn't great about this, I think the proper thing to do is add a protected accessor rather than make the member variable protected, and add a unit test so someone (me) doing coverage sweeps doesn't revert it =P

Reply:
Thanks Alyssa for reviewing 💯 .  I will add a unit test to cover the case. The reason for change was to allow custom Sockets access to the callbacks (and `Connection` via the callbacks)..